react-app/src/components/Common/sidebar.js

```diff
@@ -1,12 +1,30 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
 import 'antd/dist/antd.css';
 import '../../index.css';
 import { Layout, Menu } from 'antd';
-import NewButton from './newbutton';
+import NewButton  from './newbutton';
+import axios from "axios";
+import { useSelector } from 'react-redux';
 
 const { Sider } = Layout;
 
 const Sidebar = () => {
+
+  const currentUser = useSelector(state => state.users.currentUser)
+
+  const [menuItems, setMenuItems] = useState([])
+
+  useEffect(() => {
```

Comment:
Give it a second param otherwise it will try to fetch every time the state updated.